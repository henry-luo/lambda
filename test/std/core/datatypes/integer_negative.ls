// Test: Integer Negative Numbers
// Category: core/datatypes
// Type: positive
// Expected: -5

// Test negative integer operations
let a = -10
a
let b = 5
b
a + b
