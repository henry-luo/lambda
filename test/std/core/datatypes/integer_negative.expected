09:15:42 [NOTICE] Start transpiling test/std/core/datatypes/integer_negative.ls...
[script:list]
  [content:list[3]]
    no declare
    item:
    [let stam:null]
      declare:
      [assign expr:a:int]
        [primary expr:int,const:1]
          [const: int]
    item:
    [let stam:null]
      declare:
      [assign expr:b:int]
        [primary expr:int,const:1]
          [const: int]
    item:
    [binary expr +.3:int]
      [primary expr:int,const:1]
        [ident:a:int,const:1]
      [primary expr:int,const:1]
        [ident:b:int,const:1]
09:15:42 [NOTICE] compiling C code in 'test/std/core/datatypes/integer_negative.ls' to MIR
C2MIR init end           -- 68 usec
test/std/core/datatypes/integer_negative.ls:1:2: warning -- unknown pragma
    preprocessor tokens -- 2427, parse tokens -- 1459
  C2MIR preprocessor end    -- 506 usec
  C2MIR parser end          -- 628 usec
  C2MIR context checker end -- 706 usec
  C2MIR generator end       -- 917 usec
C2MIR compiler end                -- 947 usec
09:15:42 [NOTICE] Generating native code...
##### Script 'test/std/core/datatypes/integer_negative.ls' executed: #####
-5
